### PR TITLE
Add variable mediator docs

### PR DESCRIPTION
**Pull request by SanojPunchihewa**
## Purpose
Add variable mediator docs

localhost_8000_en_latest_reference_mediators_variable-mediator_

## Related PR
https://github.com/wso2/wso2-synapse/pull/2225

### CODE REVIEW

File: en/docs/reference/mediators/variable-mediator.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
+# Variable mediator
+
+The Variable mediator is used to manage variables within a mediation flow. The Variable Mediator allows you to set or remove variables, which can then be used later in the mediation flow.
```
Comment (chanikag) [inline en/docs/reference/mediators/variable-mediator.md:3]:
```suggestion
The Variable mediator is used to manage variables within a mediation flow. The Variable mediator allows you to set or remove variables, which can then be used later in the mediation flow.
```

**Comment by chanikag** [comment]
Let's add this to https://mi.docs.wso2.com/en/latest/reference/mediators/about-mediators/ too

File: en/docs/reference/mediators/variable-mediator.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+<td><strong>Value</strong></td>
+<td>You can give either a static value or a <a href="{{base_path}}/reference/synapse-properties/synapse-expressions">Synapse expression</a>.
+</td>
+</tr>
+</tbody>
+</table>
+
+## Examples
+
+### Set a Variable with a Literal Value
```
Comment (DinithiDiaz) [inline en/docs/reference/mediators/variable-mediator.md:65]:
```suggestion
### Set a variable with a literal value
```

File: en/docs/reference/mediators/variable-mediator.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+This example demonstrates setting a variable with a static value.
+
+```xml
+<variable name="username" action="set" value="JohnDoe" type="STRING"/>
+```
+
+### Set a Variable with an Expression
```
Comment (DinithiDiaz) [inline en/docs/reference/mediators/variable-mediator.md:73]:
```suggestion
### Set a variable with an expression
```

File: en/docs/reference/mediators/variable-mediator.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+This example demonstrates dynamically setting a variable using an expression.
+
+```xml
+<variable name="userDataObject" action="set" expression="${payload.user.data}" type="JSON"/>
+<variable name="userId" action="set" expression="${payload.user.id}" type="INTEGER"/>
+```
+
+### Remove a Variable
```
Comment (DinithiDiaz) [inline en/docs/reference/mediators/variable-mediator.md:82]:
```suggestion
### Remove a variable
```

Comment (SanojPunchihewa) [reply en/docs/reference/mediators/variable-mediator.md]: Fixed with aa3139e

File: en/docs/reference/mediators/variable-mediator.md (hunk 3 above)
Comment (SanojPunchihewa) [reply en/docs/reference/mediators/variable-mediator.md]: Fixed with aa3139e

File: en/docs/reference/mediators/variable-mediator.md (hunk 2 above)
Comment (SanojPunchihewa) [reply en/docs/reference/mediators/variable-mediator.md]: Fixed with aa3139e

File: en/docs/reference/mediators/variable-mediator.md (hunk 1 above)
Comment (SanojPunchihewa) [reply en/docs/reference/mediators/variable-mediator.md]: Fixed with aa3139e

**Comment by SanojPunchihewa** [comment]
> Let's add this to https://mi.docs.wso2.com/en/latest/reference/mediators/about-mediators/ too

Fixed with aa3139e

**Review by DinithiDiaz — APPROVED** [review]

**Pull request merged**